### PR TITLE
avoid cache concurrency situations

### PR DESCRIPTION
**Pull request by sjanssen2**
This PR addresses #60 , when two processes trigger same instance build

### CODE REVIEW

File: bellmanscafe/tests/test_execute.py [hunk 1]
```diff
@@ -1,34 +1,59 @@
...
-
-        obs = compile_and_run_gapc(gapl_programs, user_input, settings)
-        print(obs)
+        obs = compile_and_run_gapc(
+            self.gapl_programs, self.user_input, self.settings)
+        self.assertEqual('( -3 , 4 )', obs['run']['stdout'][-1].strip())
+
+    def test_concurrency(self):
+        ERRMSG = ('looks like another process is trying to build '
+                  'the same instan')
```
Comment (Anna-Rehm) [inline bellmanscafe/tests/test_execute.py:38]: typo "instace"

File: bellmanscafe/execute.py [hunk 2]
```diff
@@ -131,11 +132,16 @@ def compile_and_run_gapc(gapl_programs, user_input, settings,
         # 3) compiling c++ into binary
         'make': {'cmds': 'make -f out.mf'},
 
-        # 4) run the compiled binary with user input (not cached)
+        # 4) create a file to indicate that binary execution (or abortion) is
+        #    complete. This avoids concurrency issues, where two users are
+        #    tiggering execution of the same instance
```
Comment (kmaibach) [inline bellmanscafe/execute.py:137]: I believe you mean `triggering`?

**Review by kmaibach — APPROVED** [review]

File: bellmanscafe/tests/test_execute.py (hunk 1 above)
Comment (sjanssen2) [reply bellmanscafe/tests/test_execute.py]: not really, I am checking here if the error message does contain the given text. `instan` is part of it :-) But you are right, more easy to read is `instance`

File: bellmanscafe/execute.py (hunk 2 above)
Comment (sjanssen2) [reply bellmanscafe/execute.py]: you are right ... and I figure I was tiggering at this moment: https://www.urbandictionary.com/define.php?term=tiggering

**Pull request merged**